Contribution guide part I (#357)
MicrosoftDocs/quantum-docs-pr: articles/contributing/index.md

---
title: Contributing to the Quantum Development Kit | Microsoft Docs
description: Contributing to the Quantum Development Kit
author: cgranade
ms.author: chgranad
ms.date: 10/12/2018
ms.topic: article
uid: microsoft.quantum.contributing
---

# Contributing to the Quantum Development Kit #

The Quantum Development Kit is more than a collection of tools for writing quantum programs.

> It's part of a broad community of people discovering quantum computing, performing research in quantum algorithms, developing new applications for quantum devices, and otherwise working to make the most out of the quantum programming.

Reviewer: Software (or hardware) is typically not considered part of communities of people.

- **Improving documentation.** Any documentation set can always be better, can cover more details, be made more accessible.
- **Contributing code.** Of course, one of the most direct ways to contribute is by adding new code to the Quantum Development Kit.

These different kinds of contributions are all immensely valuable, and are greatly appreciated.
In the rest of the guide, we'll offer advice on how to make each kind of contribution.

## Where Do Contributions Go? ##

The Quantum Development Kit includes a number of different pieces that all work together to realize a platform for writing quantum programs.
Each of these different pieces finds its home on a different repository, so the one of the earlier things to sort out is where each contribution best belongs.

- [**Microsoft/Quantum**](https://github.com/Microsoft/Quantum): Samples and tools to help get started with the Quantum Development Kit.
- [**Microsoft/QuantumLibraries**](https://github.com/Microsoft/QuantumLibraries): Standard and domain-specific libraries for the Quantum Development Kit.

> - [**Microsoft/QuantumKatas**](https://github.com/Microsoft/QuantumKatas): Self-paced exercises to help learn more about quantum programming and the Q# language.

Reviewer: ```suggestion
- [**Microsoft/QuantumKatas**](https://github.com/Microsoft/QuantumKatas): Self-paced programming exercises for learning quantum computing and the Q# programming language.
```

- [**MicrosoftDocs/quantum-docs-pr**](https://github.com/MicrosoftDocs/quantum-docs-pr): Source code for the documentation published at https://docs.microsoft.com/quantum.

There are also a few other, more specialized repositories focusing on different events, or on auxillary functionality related to the Quantum Development Kit.

> [!NOTE]
> We unfortunately cannot accept code and documentation contributions on the [**Microsoft/Quantum-NC**](https://github.com/Microsoft/Quantum-NC) repository at this time, but we still very much appreciate bug reports.

## Next steps ##

Thanks for being a part of the Quantum Development Kit community, we're excited for your contributions!
If you'd like to learn more about contributing, please continue with one of the following guides.

<div class="nextstepaction">
[Learn how to report bugs](xref:microsoft.quantum.contributing.reporting)
</div>
<div class="nextstepaction">
[Learn how to contribute documentation](xref:microsoft.quantum.contributing.docs)
</div>
<div class="nextstepaction">
[Learn how to open pull requests](xref:microsoft.quantum.contributing.pulls)
</div>

> 

Reviewer: Should there be a link to contributing code?
